src/SimpleFrontendServer/SimpleFrontendServer.h

```diff
@@ -21,21 +22,51 @@ namespace fs = std::filesystem;
 
 namespace carta {
 #define HTTP_200 "200 OK"
+#define HTTP_400 "400 Bad Request"
 #define HTTP_404 "404 Not Found"
+#define HTTP_403 "403 Forbidden"
 #define HTTP_500 "500 Internal Server Error"
+#define HTTP_501 "501 Not Implemented"
+
+typedef uWS::HttpRequest Req;
+typedef uWS::HttpResponse<false> Res;
 
 class SimpleFrontendServer {
 public:
-    SimpleFrontendServer(fs::path root_folder);
+    SimpleFrontendServer(fs::path root_folder, std::string auth_token);
     bool CanServeFrontend() {
         return _frontend_found;
     }
-    void HandleRequest(uWS::HttpResponse<false>* res, uWS::HttpRequest* req);
+
+    void RegisterRoutes(uWS::App& app);
+    nlohmann::json GetExistingPreferences();
```

Comment:
These methods are public because they're being tested, rather than the handlers (which would require mocking the `uWS::App`, `uWS::HttpRequest` and `uWS::HttpResponse` classes.

Reply:
One option is to do the following:

```
// Allows testing of protected methods in SimpleFrontendServer without polluting the original class
class TestSimpleFrontendServer : public carta::SimpleFrontendServer {
public:
    TestSimpleFrontendServer(fs::path root_folder, std::string auth_token):
        carta::SimpleFrontendServer(root_folder, auth_token)
    {

    }
    FRIEND_TEST(RestApiTest, UpdatePreferencesFromString);
    FRIEND_TEST(RestApiTest, EmptyStartingPrefs);
    FRIEND_TEST(RestApiTest, GetExistingPrefs);
    FRIEND_TEST(RestApiTest, DeletePrefsEmpty);
    FRIEND_TEST(RestApiTest, DeletePrefsInvalid);
    FRIEND_TEST(RestApiTest, DeletePrefsIgnoresInvalidKeys);
    FRIEND_TEST(RestApiTest, DeletePrefsHandlesMissingKeys);
    FRIEND_TEST(RestApiTest, DeletePrefsSingleKey);
    FRIEND_TEST(RestApiTest, DeletePrefsKeyList);
    FRIEND_TEST(RestApiTest, EmptyStartingLayouts);
    FRIEND_TEST(RestApiTest, GetExistingLayouts);
    FRIEND_TEST(RestApiTest, DeleteLayout);
    FRIEND_TEST(RestApiTest, DeleteLayoutEmpty);
    FRIEND_TEST(RestApiTest, DeleteLayoutInvalid);
    FRIEND_TEST(RestApiTest, DeleteLayoutIgnoresInvalidKeys);
    FRIEND_TEST(RestApiTest, DeleteLayoutMissingName);
    FRIEND_TEST(RestApiTest, SetLayout);
};
```

The methods can then be made `protected` instead of `public`

Reply:
I think it's OK to add the `FRIEND_TEST`s for now -- if we can figure out a cleaner way later, we can change it.